include/ntcore_c.h

```diff
@@ -283,67 +491,107 @@ struct NT_EntryInfo* NT_GetEntryInfo(const char* prefix, size_t prefix_len,
  */
 void NT_Flush(void);
 
+/** Flush Entries.
+ * Forces an immediate flush of all local entry changes to network.
+ * Normally this is done on a regularly scheduled interval (see
+ * NT_SetUpdateRate()).
+ *
+ * Note: flushes are rate limited to avoid excessive network traffic.  If
+ * the time between calls is too short, the flush will occur after the minimum
+ * time elapses (rather than immediately).
+ *
+ * @param inst      instance handle
+ */
+void NT_Flush2(NT_Inst inst);
+
 /*
  * Callback Creation Functions
  */
 
 void NT_SetListenerOnStart(void (*on_start)(void* data), void* data);
+void NT_SetListenerOnStart2(NT_Inst inst, void (*on_start)(void* data),
```

Comment:
The new listeners that take an instance should also return the instance in the callback, so the instance can be tetermined.